### DISCUSSION TITLE
DisallowHasFactorySniff Why?

### DISCUSSION THREAD

**Discussion by sojeda** [worksome/coding-style#38] [Q&A] [answered]
The package is very good, thank you very much

I would like to understand why this rule exists: DisallowHasFactorySniff

` Models should not use the `HasFactory` trait.
`

**Comment by olivernybroe** [comment, marked as answer]
@sojeda Sure, will gladly explain why 🙂 

The reason is that factories should be used in tests. Adding HasFactory to your model is adding code which is meant for testing into your application code.

Essentially you end up having code which will never get called outside of tests in every single model. 


However it can be a bit annoying having to use the longer syntax in tests. 👀


I see three possible solutions here
- remove the check and live with the consequence
- create a PR for Laravel to remove the need for HasFactory trait. (Maybe macro's)
- or you can disable this check in your codebase

**Reply by sojeda** [reply]
Thanks for the reply!
